lib/spack/spack/binary_distribution.py

```diff
@@ -239,23 +239,39 @@ def sign_tarball(key, force, specfile_path):
     Gpg.sign(key, specfile_path, '%s.asc' % specfile_path)
 
 
-def generate_index(outdir, indexfile_path):
-    f = open(indexfile_path, 'w')
+def generate_html_index(pathList, outputPath):
+    f = open(outputPath, 'w')
     header = """<html>\n
 <head>\n</head>\n
 <list>\n"""
     footer = "</list>\n</html>\n"
-    paths = os.listdir(outdir + '/build_cache')
     f.write(header)
-    for path in paths:
+    for path in pathList:
         rel = os.path.basename(path)
         f.write('<li><a href="%s"> %s</a>\n' % (rel, rel))
     f.write(footer)
     f.close()
 
 
+def generate_yaml_index(pathList, outputPath):
```

Comment:
this is a good start but likely needs a schema like the other YAML files.  Also, I think `json` is actually better for things to be read by computers (not people).  Does your `checkbinaries` command actually look at this YAML file or just the HTML?

Reply:
I'll change it to `json`, do we already have json schemas somewhere in the repo?  Once I have it change to `json`, I'll update the `check-binaries` code to read that instead.